From c950667e87dc175bd2741a51460ebd3e29a57c92 Mon Sep 17 00:00:00 2001
From: Vincent Abriou <vincent.abriou@st.com>
Date: Wed, 5 Oct 2016 16:14:07 +0200
Subject: [PATCH] libweston: include weston-egl-ext.h in drm, x11 and wayland
 compositor

As to what is done for gl-renderer.c, weston-egl-ext.h should be
include in compositor-drm.c, compositor-x11.c and compositor-wayland.c.
This fix building issue with GPU that does not have EGL_PLATFORM_xxx_KHR
in their extension header file eglext.h.

Signed-off-by: Vincent Abriou <vincent.abriou@st.com>
Reviewed-by: Daniel Stone <daniels@collabora.com>
---
 libweston/compositor-drm.c     | 1 +
 libweston/compositor-wayland.c | 1 +
 libweston/compositor-x11.c     | 1 +
 3 files changed, 3 insertions(+)

diff --git a/libweston/compositor-drm.c b/libweston/compositor-drm.c
index 567240f..f61e3d9 100644
--- a/libweston/compositor-drm.c
+++ b/libweston/compositor-drm.c
@@ -52,6 +52,7 @@
 #include "shared/helpers.h"
 #include "shared/timespec-util.h"
 #include "gl-renderer.h"
+#include "weston-egl-ext.h"
 #include "pixman-renderer.h"
 #include "libbacklight.h"
 #include "libinput-seat.h"
diff --git a/libweston/compositor-wayland.c b/libweston/compositor-wayland.c
index 580c7b5..46fdde1 100644
--- a/libweston/compositor-wayland.c
+++ b/libweston/compositor-wayland.c
@@ -44,6 +44,7 @@
 #include "compositor.h"
 #include "compositor-wayland.h"
 #include "gl-renderer.h"
+#include "weston-egl-ext.h"
 #include "pixman-renderer.h"
 #include "shared/helpers.h"
 #include "shared/image-loader.h"
diff --git a/libweston/compositor-x11.c b/libweston/compositor-x11.c
index dadcd10..1c6de08 100644
--- a/libweston/compositor-x11.c
+++ b/libweston/compositor-x11.c
@@ -56,6 +56,7 @@
 #include "shared/helpers.h"
 #include "shared/image-loader.h"
 #include "gl-renderer.h"
+#include "weston-egl-ext.h"
 #include "pixman-renderer.h"
 #include "presentation-time-server-protocol.h"
 #include "linux-dmabuf.h"
-- 
2.7.4

